phpBB/phpbb/notification/type/mention.php

```diff
@@ -0,0 +1,407 @@
+
+    /** @var \phpbb\user_loader */
+    protected $user_loader;
+
+    /** @var \phpbb\config\config */
+    protected $config;
+
+    /**
+    * Inherit notification read status from post.
+    *
+    * @var bool
+    */
+    protected $inherit_read_status = true;
+
+    static public $notification_option = false;
+
+    public function set_config(\phpbb\config\config $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+    * Initiate the class member with the object of class \phpbb]user_loader.
+    *
+    */
+    public function set_user_loader(\phpbb\user_loader $user_loader)
+    {
+        $this->user_loader = $user_loader;
+    }
+
+    /**
+    * Is available
+    */
+    public function is_available()
+    {
+        return $this->config['allow_topic_notify'];
+    }
+
+    /**
+    * Get the id of the item
+    *
+    * @param array $post The data from the post
+    * @return int The post id
+    */
+    static public function get_item_id($post)
+    {
+        return (int) $post['post_id'];
+    }
+
+    /**
+    * Get the id of the parent
+    *
+    * @param array $post The data from the post
+    * @return int The topic id
+    */
+    static public function get_item_parent_id($post)
+    {
+        return (int) $post['topic_id'];
+    }
+
+    /**
+    * Get the user's avatar
+    */
+    public function get_avatar()
+    {
+        return $this->user_loader->get_avatar($this->get_data('poster_id'), false, true);
+    }
+
+    /**
+    * Retrieves an ArrayIterator over the configuration values.
+    *
+    * @return \ArrayIterator An iterator over all config data
+    */
+    public function getIterator()
+    {
+        return new \ArrayIterator($this->config);
+    }
+
+    /**
+    * Extract user_ids of all users mentioned in the post.
+    *
+    * @return \Array An array of all mentioned userids.
+    */
+    public function find_users_for_notification($db, $users_to_notify = array())
+    {
+        if (count($users_to_notify) > 0)
+        {
+            $sql_ary = "SELECT users.user_id, users.username_clean FROM ". USERS_TABLE . " as users, " . USER_NOTIFICATIONS_TABLE . " as notif WHERE " . $db->sql_in_set('username_clean', $users_to_notify) . " AND users.user_id = notif.user_id AND notif.item_type = 'notification.type.mention' " ;
+            $result = $db->sql_query($sql_ary);
+            $user_list = array();
+            while ($row = $db->sql_fetchrow($result))
+            {
+                $user_list[$row["username_clean"]] = $row["user_id"];
+            }
+            $db->sql_freeresult($result);
+            return $user_list;
+        }
+        return array();
+    }
+
+    /**
+    * Get the type and method of notification corresponding the user
+    *
+    * @return \Array Array containing the notification type object and notification
+    * method object.
+    */
+    public function get_notification_type_and_method($db, $user_list, $notif_manager)
```

Comment:
Since, I am using add_notifications_for_users() from manager class to send notifications, this function needs an array with key value pairs where key is userid and value is the method name.
And get_notification_type_and_method() function is doing this task.